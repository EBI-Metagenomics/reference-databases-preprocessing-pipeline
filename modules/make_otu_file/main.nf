
process MAKE_OTU_FILE {

    label 'heavy'
    publishDir "${params.outdir}/${label}/${version}/", mode: 'copy'

    input:
    path tax
    path taxid
    val version
    val label

    output:
    path("*.otu"), emit: otu

    """
    case $label in

        UNITE)
            make_otu_file.py -tx $tax -l $label
        ;;
        
        PR2)
            make_otu_file.py -tx $tax -l $label --ext_ranks
        ;;

        SILVA-SSU | SILVA-LSU | ITSone)
            make_otu_file.py -tx $tax -t $taxid -l $label
        ;;

        *)
            echo "Incorrect reference db label"
            exit 1
        ;;
        
    esac
    """

}