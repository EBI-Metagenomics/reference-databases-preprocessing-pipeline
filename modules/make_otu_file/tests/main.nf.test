nextflow_process {

    name "Test Process MAKE_OTU_FILE"
    script "modules/make_otu_file/main.nf"
    process "MAKE_OTU_FILE"
    tag "modules"
    tag "make_otu_file"

    test("Test make_otu_file process for SILVA/ITSone type input") {
        tag "make_otu_file_silva-itsone"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/make_otu_file/tests/data/SILVA-ITSone/SILVA-ITSone-test-tax.txt', checkIfExists: true)
                input[1] = file('${projectDir}/modules/make_otu_file/tests/data/SILVA-ITSone/SILVA-ITSone-test.taxid', checkIfExists: true)
                input[2] = params.silva_version
                input[3] = params.silva_ssu_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.otu).match("silva-itsone_otu") }
            )
        }

    }

    test("Test make_otu_file process for UNITE type input") {
        tag "make_otu_file_unite"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/make_otu_file/tests/data/UNITE/UNITE-test-tax.txt', checkIfExists: true)
                input[1] = file('${projectDir}/modules/make_otu_file/tests/data/UNITE/EMPTY.txt', checkIfExists: true)
                input[2] = params.unite_version
                input[3] = params.unite_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.otu).match("unite_otu") }
            )
        }

    }

    test("Test make_otu_file process for PR2 type input") {
        tag "make_otu_file_pr2"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/make_otu_file/tests/data/PR2/PR2-test-tax.txt', checkIfExists: true)
                input[1] = file('${projectDir}/modules/make_otu_file/tests/data/PR2/EMPTY.txt', checkIfExists: true)
                input[2] = params.pr2_version
                input[3] = params.pr2_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.otu).match("pr2_otu") }
            )
        }

    }

}
