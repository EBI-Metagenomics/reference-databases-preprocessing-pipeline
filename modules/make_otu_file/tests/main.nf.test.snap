{
    "unite_otu": {
        "content": [
            [
                "UNITE.otu:md5,db9a79644972538f49b9e4cf38fae24f"
            ]
        ],
        "timestamp": "2023-12-12T09:57:10.013618152"
    },
    "silva-itsone_otu": {
        "content": [
            [
                "SILVA-SSU.otu:md5,62e5702a8aec344230c44d08a39d6257"
            ]
        ],
        "timestamp": "2023-12-12T09:52:44.452627697"
    },
    "pr2_otu": {
        "content": [
            [
                "PR2.otu:md5,670cabf6f22e416bb885fea312efb177"
            ]
        ],
        "timestamp": "2023-12-12T09:52:54.491946408"
    }
}